newton/_src/geometry/utils.py

```diff
@@ -31,6 +31,64 @@
 )
 
 
+# Warp kernel for inertia-based OBB computation
+@wp.kernel
+def compute_obb_candidates(
+    vertices: wp.array(dtype=wp.vec3),
+    base_quat: wp.quat,
+    num_angles_per_axis: int,
+    volumes: wp.array(dtype=float),
+    transforms: wp.array(dtype=wp.transform),
+    extents: wp.array(dtype=wp.vec3),
+):
+    """Compute OBB candidates for different rotations around principal axes."""
+    tid = wp.tid()
+
+    # Decode which axis and angle this thread handles
+    axis_idx = tid // num_angles_per_axis
+    angle_idx = tid % num_angles_per_axis
+
+    if axis_idx >= 3:
+        return
```

Comment:
I think it would be clearer to have a 2D kernel that is launched with `dim=(num_angles_per_axis, 3)` and then use a 2D tid:
```py
angle_idx, axis_idx = wp.tid()
``
Then you can also skip the axis_idx bounds check.